libs/langchain-community/src/llms/googlegenerativeai.ts

```diff
@@ -0,0 +1,380 @@
+  convertInput,
+} from "../utils/googlegenerativeai.js";
+
+import { LLM, type BaseLLMParams } from "@langchain/core/language_models/llms";
+
+export type { HarmCategory, HarmBlockThreshold, SafetySetting };
+/**
+ * Input for Text generation for Google Generative AI
+ */
+export interface GoogleGenerativeAITextInput extends BaseLLMParams {
+  /**
+   * Model Name to use
+   *
+   * Note: The format must follow the pattern - `{model}`
+   */
+  modelName?: string;
+
+  /**
+   * Controls the randomness of the output.
+   *
+   * Values can range from [0.0,1.0], inclusive. A value closer to 1.0
+   * will produce responses that are more varied and creative, while
+   * a value closer to 0.0 will typically result in more straightforward
+   * responses from the model.
+   *
+   * Note: The default value varies by model
+   */
+  temperature?: number;
+
+  /**
+   * Maximum number of tokens to generate in the completion.
+   */
+  maxOutputTokens?: number;
+
+  /**
+   * Top-p changes how the model selects tokens for output.
+   *
+   * Tokens are selected from most probable to least until the sum
+   * of their probabilities equals the top-p value.
+   *
+   * For example, if tokens A, B, and C have a probability of
+   * .3, .2, and .1 and the top-p value is .5, then the model will
+   * select either A or B as the next token (using temperature).
+   *
+   * Note: The default value varies by model
+   */
+  topP?: number;
+
+  /**
+   * Top-k changes how the model selects tokens for output.
+   *
+   * A top-k of 1 means the selected token is the most probable among
+   * all tokens in the model’s vocabulary (also called greedy decoding),
+   * while a top-k of 3 means that the next token is selected from
+   * among the 3 most probable tokens (using temperature).
+   *
+   * Note: The default value varies by model
+   */
+  topK?: number;
+
+  /**
+   * The set of character sequences (up to 5) that will stop output generation.
+   * If specified, the API will stop at the first appearance of a stop
+   * sequence.
+   *
+   * Note: The stop sequence will not be included as part of the response.
+   */
+  stopSequences?: string[];
+
+  /**
+   * A list of unique `SafetySetting` instances for blocking unsafe content. The API will block
+   * any prompts and responses that fail to meet the thresholds set by these settings. If there
+   * is no `SafetySetting` for a given `SafetyCategory` provided in the list, the API will use
+   * the default safety setting for that category.
+   */
+  safetySettings?:
+    | SafetySetting[]
+    | protos.google.ai.generativelanguage.v1beta2.ISafetySetting[];
+
+  /**
+   * Google Palm API key to use
+   */
+  apiKey?: string;
+}
+
+/**
+ * Google Generative AI Models Wrapper to generate texts
+ */
+export class GoogleGenerativeAI
+  extends LLM
+  implements GoogleGenerativeAITextInput
+{
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      apiKey: "GOOGLE_API_KEY",
+    };
+  }
```

Comment:
Should set `static lc_name` too for minified environments (though I think this won't be supported anyway)